fdbclient/BackupAgentBase.actor.cpp

```diff
@@ -739,8 +782,36 @@ ACTOR Future<int> kvMutationLogToTransactions(Database cx,
 				                          cx,
 				                          tenantMap,
 				                          provisionalProxy));
+				if (tenantMapChanging && req.transaction.mutations.size()) {
+					// If the tenantMap is changing send the previous CommitTransactionRequest to the CommitProxy
```

Comment:
This comment should explain more about _why_ this is necessary.

Reply:
We should explain what we get from a single decodeBackupLogValue, which is an original transaction, and that the original transactions on the cluster would be such that tenant creation and first use would have to occur in different transaction batches, and so during restore since batch boundaries can be more coarse we must force a batch boundary after any original transaction batch being replayed that touches the tenant metadata.